document/core/syntax/types.rst

```diff
@@ -349,6 +349,25 @@ In a :ref:`module <syntax-module>`, each member of a recursive type is assigned
 The syntax of sub types is :ref:`generalized <syntax-heaptype-ext>` for the purpose of specifying :ref:`validation <valid>` and :ref:`execution <exec>`.
 
 
+.. _index:: ! index type
+   pair: abstract syntax; index type
+   single: memory; index type
+   single: table; index type
+.. _syntax-idxtype:
+
+Index Type
```

Comment:
I have come to wonder whether "index type" is the appropriate name, since we use "index" for static indices only. Perhaps "offset type" (offtype) would be more appropriate. But we can figure that out in a later PR.

Reply:
Fair enough.  I'll open another issue for that: https://github.com/WebAssembly/memory64/issues/67